### PR TITLE
Fix H3HexagonLayer lighting

### PR DESCRIPTION
**Pull request by Pessimistress**
H3HexagonLayer's lighting effect is inconsistent with the rest of the layers.

#### Change List
- `ColumnLayer` enforces winding order if external vertices are supplied
- `H3HexagonLayer` passes `_windingOrder` to `PolygonLayer` in high-precision mode

### CODE REVIEW

File: modules/layers/src/column-layer/column-geometry.js [hunk 1]
```diff
@@ -40,18 +47,20 @@ function tesselateColumn(props) {
...
     for (let k = 0; k < 2; k++) {
-      positions[i + 0] = vertex ? vertex[0] : cos * radius;
-      positions[i + 1] = vertex ? vertex[1] : sin * radius;
+      positions[i + 0] = vertices ? vertices[vertexIndex * 2] : cos * radius;
+      positions[i + 1] = vertices ? vertices[vertexIndex * 2 + 1] : sin * radius;
       positions[i + 2] = (1 / 2 - k) * height;
 
-      normals[i + 0] = vertex ? nextVertex[0] - vertex[0] : cos;
-      normals[i + 1] = vertex ? nextVertex[1] - vertex[1] : sin;
+      normals[i + 0] = vertices ? vertices[vertexIndex * 2] - vertices[nextVertexIndex * 2] : cos;
```
Comment (igorDykhta) [inline modules/layers/src/column-layer/column-geometry.js:60]: currentVertex - nextVertex doesn't look like the direction of normal here. Is it intentional? For same config N(cos, sin) !== N(currentVertex - nextVertex)

Comment (Pessimistress) [reply modules/layers/src/column-layer/column-geometry.js:60]: Hmmm you are right, that seems to be another bug in this class. It should just be `currentVertex` (we are not doing flat shading in this layer)

File: modules/layers/src/column-layer/column-geometry.js [hunk 2]
```diff
@@ -40,18 +47,20 @@ function tesselateColumn(props) {
   //
   for (let j = 0; j < vertsAroundEdge; j++) {
```
[the remark this answers is not in the document]
Comment (Pessimistress) [reply modules/layers/src/column-layer/column-geometry.js:48]: > it increased

Can you elaborate? What exactly is more performant?

[the remark this answers is not in the document]
Comment (Pessimistress) [reply modules/layers/src/column-layer/column-geometry.js:48]: Do you mean skipping the side triangles if the layer is not extruded? We can add the toggle in a separate PR.

[the remark this answers is not in the document]
Comment (igorDykhta) [reply modules/layers/src/column-layer/column-geometry.js:48]: As I understand when height=0 > the side triangles are still processed by H3 vertex shader and are discarded before the FS. Hm, did some testing, for a 4kk h3 dataset. I have 5 FPS with our current column-geometry implementation and 8 FPS when side vertices/indices aren't generated. It was definitely more when I did some test a while ago. I guess such change can make such large projects more "interactive". Low-end systems & mobiles can benefit as well.

**Review by igorDykhta — APPROVED** [review]

**Pull request merged**